In [1]:
import pandas as pd

df = pd.read_csv('../data/items_titles_test.csv')

import sys
sys.path.append('..')
from scripts.similarity import TextSimilarityPipeline

# Crear instancia y ejecutar
similitud = TextSimilarityPipeline(df)
resultado = similitud.run_pipeline()

# Mostrar resultado
resultado

Calculando similitudes por lotes...


100%|██████████| 20/20 [00:18<00:00,  1.11it/s]


Cálculo por lotes completado en 18.26 segundos
Pipeline ejecutado en 18.49 segundos


Término 1  \
742933            Sapato Usthemp Legend Temático - Pug 3   
263902                                     Tênis Slip On    
104550       Tênis Usthemp Long Temático - Tingo Almeida   
681808     Tênis Usthemp Folks Vegano Casual Estampa Olly   
498842  Tênis Usthemp Volare Vegano Casual Estampa Joa...   
...                                                   ...   
848993              Bicicleta Aro 29 Tsw Evo Quest Gx 12v   
13934   Tenis Masculino Casual Urbano Retrô Classic Ne...   
114956  Tênis Infantil Pampili Summer Rosa Chiclete 65...   
246070  Tênis adidas Runfalcon Original Cinza E Azul G...   
536494            Tênis De Lã Italiana Yuool Panna Tam 38   

                                                Término 2  Similitud  
742933            Sapato Usthemp Legend Temático - Pug 4   1.000000  
263902                                      Tênis Slip On   1.000000  
104550      Tênis Usthemp Long Temático - Jardim Secreto   1.000000  
681808  Tênis Usthemp Folks Vegano Casual Estampa Pequ...   1.000000  
498842  Tênis Usthemp Volare Vegano Casual Estampa Carpas   1.000000  
...                                                   ...        ...  
848993  Bicicleta Serena Feminina Aro 20 Violeta Maste...   0.500001  
13934         Tênis Bom E Barato Masculino Trilupy Casual   0.500001  
114956            Tênis By Kids Carro Infantil Chk Elas     0.500000  
246070  Tênis Adrun Flex Up-feminino-8621f-1003-cinza/...   0.500000  
536494  Tênis Colcci Bia 2021 - Modelo Novo, Detalhes ...   0.500000  

[853492 rows x 3 columns]

In [2]:
df.shape

(10000, 1)

## Análisis de Resultados de Similitud de Texto

### Detalles del Dataset
- **Cantidad de textos analizados:** 10,000

### Tiempos de Ejecución
1. **Cálculo paralelo de similitudes:** 18.26 segundos.
2. **Tiempo total del pipeline:** 18.49 segundos.

### Observaciones
- El uso de **Truncated SVD** para reducir la dimensionalidad, combinado con paralelización, permitió calcular similitudes entre todos los pares de textos en un tiempo razonable para el tamaño del dataset.
- Se filtra basado en un threshold de 0.5 para solo procesar productos que tengan alguna relación.
- Este enfoque es escalable y puede manejar datasets de gran tamaño con optimizaciones adicionales.